packages/cactus-test-plugin-ledger-connector-besu/src/test/typescript/integration/plugin-validator-besu/get-block-endpoint.test.ts

```diff
@@ -0,0 +1,149 @@
+    logLevel,
+    instanceId: uuidv4(),
+  });
+
+  const httpServer1 = createServer();
+  await new Promise((resolve, reject) => {
+    httpServer1.once("error", reject);
+    httpServer1.once("listening", resolve);
+    httpServer1.listen(0, "127.0.0.1");
+  });
+  const addressInfo1 = httpServer1.address() as AddressInfo;
+  t.comment(`HttpServer1 AddressInfo: ${JSON.stringify(addressInfo1)}`);
+  const node1Host = `http://${addressInfo1.address}:${addressInfo1.port}`;
+  t.comment(`Cactus Node 1 Host: ${node1Host}`);
+
+  const besuTestLedger = new BesuTestLedger();
+  await besuTestLedger.start();
+
+  const tearDown = async () => {
+    await besuTestLedger.stop();
+    await besuTestLedger.destroy();
+  };
+
+  test.onFinish(tearDown);
+
+  const rpcApiHttpHost = await besuTestLedger.getRpcApiHttpHost();
+  const rpcApiWsHost = await besuTestLedger.getRpcApiWsHost();
+
+  // 2. Instantiate plugin registry which will provide the web service plugin with the key value storage plugin
+  const pluginRegistry = new PluginRegistry({ plugins: [keychain] });
+
+  // 3. Instantiate the web service consortium plugin
+  const options: IPluginLedgerConnectorBesuOptions = {
+    instanceId: uuidv4(),
+    rpcApiHttpHost,
+    rpcApiWsHost,
+    pluginRegistry,
+    logLevel,
+  };
+  const pluginValidatorBesu = new PluginLedgerConnectorBesu(options);
+
+  // 4. Create the API Server object that we embed in this test
+  const configService = new ConfigService();
+  const apiServerOptions = configService.newExampleConfig();
+  apiServerOptions.authorizationProtocol = AuthorizationProtocol.NONE;
+  apiServerOptions.configFile = "";
+  apiServerOptions.apiCorsDomainCsv = "*";
+  apiServerOptions.apiPort = addressInfo1.port;
+  apiServerOptions.cockpitPort = 0;
+  apiServerOptions.apiTlsEnabled = false;
+  const config = configService.newExampleConfigConvict(apiServerOptions);
+
+  pluginRegistry.add(pluginValidatorBesu);
+
+  const apiServer = new ApiServer({
+    httpServerApi: httpServer1,
+    config: config.getProperties(),
+    pluginRegistry,
+  });
+
+  // 5. make sure the API server is shut down when the testing if finished.
+  test.onFinish(() => apiServer.shutdown());
+
+  // 6. Start the API server which is now listening on port A and it's healthcheck works through the main SDK
+  await apiServer.start();
+
+  // 7. Instantiate the main SDK dynamically with whatever port the API server ended up bound to (port 0)
+  t.comment(`AddressInfo: ${JSON.stringify(addressInfo1)}`);
+
+  const request: GetBlockV1Request = {
+    blockHashOrBlockNumber: 0,
+  };
+
+  const configuration = new BesuApiClientOptions({ basePath: node1Host });
+  const api = new BesuApiClient(configuration);
+
+  // Test for 200 valid response test case
+  const res = await api.getBlockV1(request);
+
+  const { status, data } = res;
+  t.true(status >= 200, "status GTE 200 OK");
+  t.true(status < 300, "status LT 300 OK");
+  t.ok(data, "GetBlockResponse Truthy OK");
```

Comment:
@twiggins120 Please also check if the data field has the expected contents (an object with the block information should be in there)

cc: @ty-lazar

Reply:
@petermetz  All of the requested changes have been resolved in the morning checkup!